_posts/2024-09-20-Detect-data-mutation-patterns-with-Debezium.adoc

```diff
@@ -0,0 +1,172 @@
+Application Performance Metrics:: These measure the application’s operational performance to ensure the application performs optimally and meets user expectations. For example: Request Rate, Error Rate, Latency, Throughput.
+Business Metrics:: These focus on the application's impact on business goals, such as: Number of Transactions, Conversion Rate, Revenue, User Signups.
+
+Usually, infrastructure and application metrics are quite "standard," and there are frameworks/tools that can be easily used to instruct your application to expose those metrics.
+On the other hand, business metrics can vary between applications, so you need to implement custom code every time to expose those metrics.
+
+The idea is to think at a lower level and leverage change data capture (CDC) to extract data that can be used as business metrics.
+
+Suppose we have an order service that processes and stores orders made by our customers.
+An important business metric to monitor would be the number of orders made.
+This number is very relevant to the business because it is directly correlated with the company’s earnings.
+A relevant drop in number of orders should alter the team responsible for that service as soon as possible.
+
+Normally, we would need to modify our application to instruct a https://prometheus.io/docs/concepts/metric_types/#counter[counter] metric that increments every time an order is correctly placed and stored in the database.
+But what if we could extract this information just by monitoring the database through CDC?
+
+Let’s explore how Debezium offers this possibility.
+
+== How Debezium Exposes Metrics
+
+Debezium exposes three categories of metrics:
+
+Snapshot metrics:: Provide information about connector operation while performing a snapshot, such as RowsScanned, RemainingTableCount, TotalTableCount.
+Streaming metrics:: Provide information about connector operation when the connector is reading the database log, such as TotalNumberOfCreateEventsSeen, QueueRemainingCapacity, QueueTotalCapacity.
+Schema history metrics:: Provide information about the status of the connector’s schema history, such as ChangesApplied, ChangesRecovered, Status.
+
+All these metrics are exposed through JMX. You can read more in our https://debezium.io/documentation/reference/stable/operations/monitoring.html[documentation].
+
+=== Database Activity Metrics
+
+Starting from the 3.0.0.Final release (specifically 3.0.0.Beta1), the following new streaming metrics have been introduced:
+
+NumberOfCreateEventsSeen:: Counts the number of create events per table.
+NumberOfDeleteEventsSeen:: Counts the number of delete events per table.
+NumberOfUpdateEventsSeen:: Counts the number of update events per table.
+NumberOfTruncateEventsSeen:: Counts the number of truncate events per table.
+
+Since these metrics are experimental, you need to opt-in to enable them by setting the internal.advanced.metrics.enable property to true.
```

Comment:
```suggestion
Since these metrics are experimental, you need to opt-in to enable them by setting the `internal.advanced.metrics.enable` property to true.
```